activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -367,9 +367,7 @@ def convert_key(key) # :doc:
         key.kind_of?(Symbol) ? key.to_s : key
       end
 
-      def convert_value(value, for: nil) # :doc:
-        conversion = binding.local_variable_get(:for)
-
+      def convert_value(value, conversion: nil) # :doc:
```

Comment:
> But since `convert_value` is a private method, I think we can simply rename the argument, and avoid all of that.

If so, should we hide the method in the public API doc?

https://edgeapi.rubyonrails.org/classes/ActiveSupport/HashWithIndifferentAccess.html#method-i-convert_value

Reply:
#37878